platform/linux-generic/odp_crypto_ipsecmb.c

```diff
@@ -394,6 +408,208 @@ static int process_auth_zuc_eia3_param(odp_crypto_generic_session_t *session)
 	return 0;
 }
 
+static
+odp_crypto_alg_err_t snow3g_uea2_cipher_op(odp_packet_t pkt,
```

Comment:
Apart from the actual IMB macro used, this function is almost identical to zuc_eea3_cipher_op(). The same is true between auth_snow3g_uia2_gen() and auth_zuc_eia3_gen() and between auth_snow3g_uia2_check() and auth_zuc_eia3_check(). The code could be rearranged a bit to avoid the duplication.

Reply:
Thanks for your comments, in the latest version, replace duplicated functions with generic functions and added if-condition for different algorithm.

Reply:
The solution by adding if-conditions in the generic processing functions is not ideal and does not work well when more algorithms get added. I guess it will do for now, but when you add more algorithms later, you probably have to change it to something better.